### PR TITLE
fix(organizations): extend global-admin bypass to updateRole + isGlobalAdmin helper

### LINKED ISSUE #3510: fix(organizations): extend global-admin bypass to updateRole + centralize isGlobalAdmin helper
## Problem

Follow-up to #3505. Two related improvements surfaced during review that are intentionally out of scope for that PR:

1. `modules/organizations/controllers/organizations.membership.controller.js#updateRole` still requires the actor to be an org owner (`req.membership.role === OWNER`). Global platform admins hitting this handler (even when they are not a member of the target org) will get a 403. For consistency with the remove-member / delete-org bypass, admins should be able to change member roles too — notably to transfer ownership during moderation.

2. The pattern `Array.isArray(req.user?.roles) && req.user.roles.includes('admin')` is now repeated in `organizations.controller.js`, `organizations.membership.controller.js`, `organizations.middleware.js`, and `organizations.policy.js`. This should be centralized (e.g. `req.isGlobalAdmin` populated once in an auth middleware, or a small helper in `lib/helpers/auth.js`).

## Expected behaviour

- Global admin can call `PUT /api/organizations/:orgId/memberships/:memberId` with `{ role: 'owner' | 'admin' | 'member' }` and succeed even when not a member of the target org.
- A single source of truth for `isGlobalAdmin(req)` used consistently across the codebase.

## Scope note

Item 1 must preserve the belt-and-suspenders block against CASL shipping `update Membership` by mistake for non-owner org roles — the fix is to allow only global admins + owners, not to loosen CASL overall.

### PR DESCRIPTION
**Pull request by PierreBrisorgueil**
## Summary

- Fixes the same bug #3509 fixed on `remove`/`delete org` but on the `updateRole` endpoint: a global platform admin (`roles: ['admin']`) who is not a member of the target org used to get a 403 trying to change a member role (notably to transfer ownership during moderation).
- Centralizes the repeated `Array.isArray(req.user?.roles) && req.user.roles.includes('admin')` expression behind `lib/helpers/isGlobalAdmin.js`, now used by all three controller call-sites (remove-member, remove-org, updateRole).
- Preserves the belt-and-suspenders guard: non-owner org roles are still rejected regardless of CASL.

## Changes

- `lib/helpers/isGlobalAdmin.js` — new helper (default export, 1-liner predicate).
- `lib/helpers/tests/isGlobalAdmin.unit.tests.js` — 9 unit tests covering undefined, non-array roles, lookalikes, etc.
- `modules/organizations/controllers/organizations.membership.controller.js` — `updateRole` admits global admins even without a membership. `remove` uses the shared helper.
- `modules/organizations/controllers/organizations.controller.js` — `remove` uses the shared helper.
- `modules/organizations/tests/organizations.membership.controller.unit.tests.js` — adds 3 new `updateRole` tests: global admin with no membership succeeds, non-admin non-owner rejected, org-level admin (not global) rejected.
- `MIGRATIONS.md` — dated entry 2026-04-24.

## Test plan

- [x] `npm run lint` clean
- [x] `npm run test:unit` — 563/563 pass (includes new helper tests + updateRole admin tests)
- [x] `npm run test:integration` — 277/277 pass

Closes #3510

## Release Notes

* **New Features**
  * Global administrators can now manage organization roles and memberships without requiring explicit membership in the target organization.

* **Bug Fixes**
  * Improved authorization logic to ensure consistent global admin access across organization management operations.

* **Documentation**
  * Added migration documentation detailing authorization behavior changes.

* **Tests**
  * Added test coverage for global admin authorization scenarios and edge cases.

### CODE REVIEW

File: modules/organizations/controllers/organizations.membership.controller.js [hunk 1]
```diff
@@ -34,8 +35,11 @@ const list = async (req, res) => {
...
 const updateRole = async (req, res) => {
   try {
-    // Belt-and-suspenders: only owners can change roles (CASL blocks admins via no 'update Membership')
-    if (!req.membership || req.membership.role !== MEMBERSHIP_ROLES.OWNER) {
+    // Belt-and-suspenders: only org owners can change roles (CASL blocks non-owner org
+    // roles via no 'update Membership'). Global platform admins bypass the membership
+    // requirement for moderation — notably to transfer ownership on a third-party org.
+    const admin = isGlobalAdmin(req.user);
+    if (!admin && (!req.membership || req.membership.role !== MEMBERSHIP_ROLES.OWNER)) {
       return responses.error(res, 403, 'Forbidden', 'Only owners can change member roles')();
```
[the remark this answers is not in the document]
Comment (PierreBrisorgueil) [reply modules/organizations/controllers/organizations.membership.controller.js]: Renamed `admin` → `isPlatformAdmin` in `updateRole` to distinguish the global/platform role from the nearby `MEMBERSHIP_ROLES.ADMIN`. Fixed in cd17553fa5b26ac1a3e89b6b5620840fd5289510.

[the remark this answers is not in the document]
Comment (PierreBrisorgueil) [reply modules/organizations/controllers/organizations.membership.controller.js]: Updated the 403 message to "Only owners or global admins can change member roles". Fixed in cd17553fa5b26ac1a3e89b6b5620840fd5289510.

File: modules/organizations/controllers/organizations.controller.js [hunk 2]
```diff
@@ -107,9 +108,9 @@ const remove = async (req, res) => {
     // UX protection: prevent a regular user from deleting their own last organization.
     // Global platform admins bypass this entirely (moderation); a member of multiple orgs
     // is also safe to delete the current one since they keep at least one membership.
-    const isGlobalAdmin = Array.isArray(req.user?.roles) && req.user.roles.includes('admin');
+    const admin = isGlobalAdmin(req.user);
     const isMemberOfTarget = !!req.membership;
-    if (!isGlobalAdmin && isMemberOfTarget) {
+    if (!admin && isMemberOfTarget) {
```
[the remark this answers is not in the document]
Comment (PierreBrisorgueil) [reply modules/organizations/controllers/organizations.controller.js]: Renamed `admin` → `isPlatformAdmin` in `organizations.controller.js#remove` for the same reason. Fixed in cd17553fa5b26ac1a3e89b6b5620840fd5289510.

File: modules/organizations/controllers/organizations.membership.controller.js [hunk 3]
```diff
@@ -56,10 +60,10 @@ const remove = async (req, res) => {
   try {
     // Only owners can remove anyone; admins can only remove members.
     // Global platform admins bypass org-level RBAC for moderation needs.
-    const isGlobalAdmin = Array.isArray(req.user?.roles) && req.user.roles.includes('admin');
+    const admin = isGlobalAdmin(req.user);
     const actorRole = req.membership?.role;
     const targetRole = req.membershipDoc.role;
-    const canRemove = isGlobalAdmin
+    const canRemove = admin
       || actorRole === MEMBERSHIP_ROLES.OWNER
```
[the remark this answers is not in the document]
Comment (PierreBrisorgueil) [reply modules/organizations/controllers/organizations.membership.controller.js:71]: Renamed `admin` → `isPlatformAdmin` in the `remove` handler for consistency. Fixed in cd17553fa5b26ac1a3e89b6b5620840fd5289510.

File: modules/organizations/tests/organizations.membership.controller.unit.tests.js [hunk 4]
```diff
@@ -89,6 +89,52 @@ describe('Membership controller unit tests:', () => {
...
+      const res = mockRes();
+
+      await membershipController.updateRole(req, res);
+
+      expect(mockUpdateRole).toHaveBeenCalledTimes(1);
+      expect(mockUpdateRole).toHaveBeenCalledWith(req.membershipDoc, MEMBERSHIP_ROLES.OWNER);
+      expect(res.status).not.toHaveBeenCalledWith(403);
+    });
+
+    test('should still reject non-admin non-owner even with explicit role member', async () => {
```
[the remark this answers is not in the document]
Comment (PierreBrisorgueil) [reply modules/organizations/tests/organizations.membership.controller.unit.tests.js]: Refactored the two `updateRole` rejection regression tests to use `test.each`. Fixed in cd17553fa5b26ac1a3e89b6b5620840fd5289510.

File: modules/organizations/controllers/organizations.membership.controller.js [hunk 5]
```diff
@@ -56,10 +60,10 @@ const remove = async (req, res) => {
   try {
     // Only owners can remove anyone; admins can only remove members.
     // Global platform admins bypass org-level RBAC for moderation needs.
-    const isGlobalAdmin = Array.isArray(req.user?.roles) && req.user.roles.includes('admin');
+    const admin = isGlobalAdmin(req.user);
```
[the remark this answers is not in the document]
Comment (PierreBrisorgueil) [reply modules/organizations/controllers/organizations.membership.controller.js]: Renamed `admin` → `isPlatformAdmin` across all three controller call-sites. Fixed in cd17553fa5b26ac1a3e89b6b5620840fd5289510.

File: lib/helpers/isGlobalAdmin.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+ * @desc True when the authenticated user has the global 'admin' role.
+ *   Use sparingly — prefer CASL abilities for per-resource checks. This
+ *   helper is only for belt-and-suspenders guards inside controllers that
+ *   need to bypass org-scoped membership checks (moderation endpoints).
+ * @param {Object} user - req.user object (may be undefined)
+ * @returns {boolean}
+ */
+const isGlobalAdmin = (user) => Array.isArray(user?.roles) && user.roles.includes('admin');
+
+export default isGlobalAdmin;
```
[the remark this answers is not in the document]
Comment (PierreBrisorgueil) [reply lib/helpers/isGlobalAdmin.js:14]: Thanks — the four policy-level duplicates are intentionally out of scope for this PR (documented in MIGRATIONS.md). Tracking for a dedicated follow-up since those sit in policies, not controllers, and would require a different test surface.
